File: backend/app/constants/version_types.py
from __future__ import annotations

import enum


class VersionType(str, enum.Enum):
    app_release = "app_release"
    hotfix = "hotfix"


REVIEW_NODE_KEYS: frozenset[str] = frozenset(
    {
        "gp_review",
        "as_review",
        "website_link",
        "gp_approved",
        "as_approved",
    }
)

VERSION_TYPE_LABELS: dict[str, str] = {
    "app_release": "应用发版",
    "hotfix": "热修",
}

VERSION_TYPES: tuple[str, ...] = ("app_release", "hotfix")

File: frontend/src/constants/versionTypes.ts
import type { VersionType } from '@/types/business';

export const VERSION_TYPE_LABELS: Record<VersionType, string> = {
  app_release: '应用发版',
  hotfix: '热修',
};

export const VERSION_TYPE_OPTIONS = (Object.keys(VERSION_TYPE_LABELS) as VersionType[]).map(
  (value) => ({
    label: VERSION_TYPE_LABELS[value],
    value,
  })
);

export function versionTypeLabel(type: VersionType | string): string {
  return VERSION_TYPE_LABELS[type as VersionType] ?? type;
}
